### PR TITLE
Remove bcl2fastq sequencing parsers

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Removed all logic regarding parsing sequencing metrics from `bcl2fastq` flow cells and simplified the module hierarchy .

### Changed

- Removed the bcl converter parameter from all the finish functions
- Deleted functions, modules, tests and fixtures of sequencing metric parsing of bcl2fastq
- Moved the logic for parsing BCLConvert sequencing metrics directly to the sequencing metrics API in `cg/apps/sequencing_metrics_parser/api.py` (before this API evaluated the bcl converter and called the respective parser).
- Moved and refactored the tests for BCLConvert sequencing metrics parsing to the API tests
- Moved some fixtures to fixture plugins

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b remove-bcl2fastq-04 -a
    ```

### How to test

- [x] `cg demultiplex finish flow-cell` with bcl2fastq flow cells, verify that it fails
```shell
cg demultiplex finish flow-cell 180508_ST-E00269_0269_AHL32LCCXY
Running cg demultiplex.
Finish flow cell 180508_ST-E00269_0269_AHL32LCCXY
Samplesheet passed validation
Flow cell 180508_ST-E00269_0269_AHL32LCCXY will be skipped: No fastq files were found for any sample in flow cell HL32LCCXY
```
- [x] `cg demultiplex finish flow-cell` with BCLConvert flow cells, verify that it works
```shell
flow cell is finished correctly
```

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by SA
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MINOR** - when you add functionality in a backwards compatible manner

## Implementation Plan

- [ ] Deployed to stage:
```shell

```
- [ ] Deployed to production:
```shell

```

### CODE REVIEW

File: cg/apps/sequencing_metrics_parser/api.py [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
 ) -> list[SampleLaneSequencingMetrics]:
-    """Parse the sequencing metrics data for the correct demultiplexing software
-    into the sequencing statistics model."""
-    if bcl_converter == BclConverter.BCLCONVERT:
-        return create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell(
-            flow_cell_directory
-        )
-    return create_bcl2fastq_metrics(flow_cell_directory)
+    """Parse the demultiplexing metrics data into the sequencing statistics model."""
+    metrics_parser = BclConvertMetricsParser(flow_cell_directory)
```
Comment (diitaz93) [inline cg/apps/sequencing_metrics_parser/api.py:15]: I basically pasetd the content of `create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell` here

File: cg/apps/sequencing_metrics_parser/api.py [hunk 2]
```diff
@@ -33,13 +33,44 @@ def create_undetermined_non_pooled_metrics(
     non_pooled_lanes_and_samples: list[tuple[int, str]] = (
         flow_cell.sample_sheet.get_non_pooled_lanes_and_samples()
     )
+    metrics_parser = BclConvertMetricsParser(flow_cell.path)
```
Comment (diitaz93) [inline cg/apps/sequencing_metrics_parser/api.py:36]: I pasted the content of `create_bcl_convert_undetermined_metrics` here

File: cg/apps/sequencing_metrics_parser/api.py [hunk 3]
```diff
@@ -33,13 +33,44 @@ def create_undetermined_non_pooled_metrics(
...
         )
-    return create_bcl_convert_undetermined_metrics(
-        flow_cell_dir=flow_cell.path,
-        non_pooled_lane_sample_pairs=non_pooled_lanes_and_samples,
+        metrics.sample_internal_id = sample_internal_id
+        undetermined_metrics.append(metrics)
+    return undetermined_metrics
+
+
+def _create_bcl_convert_sequencing_metrics(
```
Comment (diitaz93) [inline cg/apps/sequencing_metrics_parser/api.py:53]: moved this function here from `cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py`

File: tests/apps/sequencing_metrics_parser/parsers/test_bcl_convert_to_sequencing_statistics.py [hunk 4]
```diff
@@ -1,72 +0,0 @@
...
-    """Test creating undetermined sequencing statistics from bcl convert metrics without undetermined reads."""
-
-    # GIVEN a directory with a flow cell demultiplexed with bcl convert without undetermined reads in a lane
-
-    # WHEN creating undetermined sequencing statistics from bcl convert metrics specifying an existing lane without undetermined reads
-    metrics: list[SampleLaneSequencingMetrics] = create_bcl_convert_undetermined_metrics(
-        flow_cell_dir=bcl_convert_metrics_dir_path,
-        non_pooled_lane_sample_pairs=[(2, "sample_id")],
-    )
-
```
Comment (diitaz93) [inline tests/apps/sequencing_metrics_parser/parsers/test_bcl_convert_to_sequencing_statistics.py:70]: moved all this functions to the module just below ⬇️

**Review by seallard — APPROVED** [review]
👍

**Pull request merged**